torch/nn/parallel/distributed.py

```diff
@@ -568,6 +573,15 @@ def no_sync(self):
             self.require_backward_grad_sync = old_require_backward_grad_sync
 
     def forward(self, *inputs, **kwargs):
+        if self.ddp_join_enabled:
+            ones = torch.ones(
+                1, device=self.device_ids[0] if self.device_type != "cpu" else "cpu"
```

Comment:
does `device = list(self.module.parameters())[0].device` work here?

Reply:
It should work, but the reason I wasn't using it is because I wasn't sure if we want the overhead of creating the list only tot take the device_id of the first parameter. Maybe we could do this once in the constructor and use that as the device for the process wherever we need it?

Reply:
Ah, I see. Yep, doing it in the ctor sounds good to me.